### ISSUE TITLE
ANN101 False Positive

### ISSUE THREAD

**Issue by Tomperez98** [charliermarsh/ruff#1231]
```python
class AClass:
    """Loan Service exposes queries and commands required by the system."""

    def public_method(self, properties: Dict[Any, Any]) -> polars.DataFrame:
        """A Docstring."""
        _ = properties

```

```bash
ANN101 Missing type annotation for `self` in method
```

**Comment by charliermarsh** [comment]
Weird, definitely a bug!

**Comment by Tomperez98** [comment]
If I write the code as

```python
class AClass:
    """Loan Service exposes queries and commands required by the system."""

    def public_method(self: "AClass", properties: Dict[Any, Any]) -> polars.DataFrame:
        """A Docstring."""
        _ = properties
```

It doesn't complain. Maybe that's the correct way of write it, but pretty shitty if it is

**Comment by edgarrmondragon** [comment]
That's the behavior of [flake8-annotations](https://pypi.org/project/flake8-annotations/). The rationale is that inheritance patterns may depend on the concrete type of `self`.

```python
class A:
    def public_method(self: "AClass", properties: Dict[Any, Any]) -> polars.DataFrame:
        """A Docstring."""
        _ = properties


class B:
    def public_method(self: "AClass", properties: Dict[Any, Any]) -> polars.DataFrame:
        """A Docstring."""
        _ = properties
```

**Comment by charliermarsh** [comment]
Wait, sorry, this actually _is_ intended, because there are special for `self` and `cls`.

**Comment by charliermarsh** [comment]
It's best to disable `ANN101` and `ANN102` if you don't want to enforce those.

**Comment by Tomperez98** [comment]
Thanks @edgarrmondragon 

Btw, is amazing what you're doing with the Meltano project
